modules/ombi.py

```diff
@@ -0,0 +1,540 @@
+
+    @cherrypy.tools.json_out()
+    @cherrypy.expose()
+    @require()
+    def tv_request_details(self,id):
+        u = 'api/v1/Request/tv/%s' % id
+        # logger.debug('Fetching request details via %s' % u)
+        d = self._ombi_get(u)
+        if d != 'False':
+            return d
+        else:
+            logger.error('Unable to get request details')
+            return 'False'
+
+    @cherrypy.tools.json_out()
+    @cherrypy.expose()
+    @require()
+    def get_tvdetails(self,id,l='request'):
+        if l == 'request':
+            u = 'api/v1/Request/tv/%s' % id
+        elif l == 'search':
+            u = 'api/v1/Search/tv/info/%s' % id
+        else:
+            logger.error('Bad request: id=%s l=%s' % (id,l))
+            return 'False'
+        # logger.debug('Fetching %s details via %s' % (l,u))
+        d = self._ombi_get(u)
+        if d != 'False':
+            return d
+        else:
+            logger.error('Unable to get tvshow details for %s' % id)
+            return 'False'
+
+    @cherrypy.tools.json_out()
+    @cherrypy.expose()
+    @require()
+    def get_searchresult(self, t, q, l):
+        logger.debug('Doing %s for %s based on %s' % (l,t,q))
+        u = ''
+        if l == 'search':
+            u = 'api/v1/Search/'+t+'/'+q
+        else:
+            if t == 'movie':
+                if q in ['popular','nowplaying','toprated','upcoming']:
+                    u = 'api/v1/Search/'+t+'/'+q
+                else:
+                    u = 'api/v1/Search/'+t+'/'+q+'/similar'
+            if t == 'tv':
+                if q in ['popular','anticipated','mostwatched','trending']:
+                    u = 'api/v1/Search/'+t+'/'+q
+        if u != '':
+            d = self._ombi_get(u)
+            if d != 'False':
+                return d
+        logger.debug('Bad %s hint' % l)
+        return 'False'
+
+    @cherrypy.tools.json_out()
+    @cherrypy.expose()
+    @require()
+    def get_extrainfo(self, t, q, k=''):
+        u = 'api/v1/Search/'+t+'/info/'+q
+        d = self._ombi_get(u)
+        if d != 'False':
+            if k == '':
+                return d
+            else:
+                return d[k]
+        logger.debug('Bad question: %s %s' % (q,k))
+        return '' # return empty string on error, in case real return value is 'False' :)
+
+    @cherrypy.expose()
+    @require()
+    @cherrypy.tools.json_out()
+    def request_movie(self, id, **kwargs):
+        logger.debug('Requesting movie id %s' % id)
+        u = 'api/v1/Request/movie'
+        d = dict()
+        d.update({ 'theMovieDbId':id })
+        r = self._ombi_post(u, d)
+        if r != 'False':
+            return r.json()
+        logger.debug('Bad request')
+        return 'False'
+
+    @cherrypy.expose()
+    @require()
+    @cherrypy.tools.json_out()
+    def request_tv(self, id, slist, spec, sel):
+        logger.debug('Requesting tvshow id %s %s' % (id,spec) )
+        d = '{ "tvDbId": %s' % int(id)
```

Comment:
Is there no other way to do this? seems to strange to create this as a string.

Reply:
I couldn't come up with a way to do this better at the time.  The constraint is that the keys in the { key: value } pair are not unique - "seasonId" and "episodes" is repeated over and over.  Everything I tried with proper dictionary objects would replace the previous items instead of append to them.

I'm not sure if my google-fu was off that night, but I see several options when googling the problem this morning, by constructing the dict using lists.  I'll have a go at rewriting this (lol fourth attempt)